tasmota/xdrv_01_webserver.ino

```diff
@@ -2114,36 +2111,46 @@ void HandleWifiConfiguration(void)
           }
         }
 
-        // remove duplicates ( must be RSSI sorted )
-        String cssid;
-        for (uint32_t i = 0; i < n; i++) {
-          if (-1 == indices[i]) { continue; }
-          cssid = WiFi.SSID(indices[i]);
-          uint32_t cschn = WiFi.channel(indices[i]);
-          for (uint32_t j = i + 1; j < n; j++) {
-            if ((cssid == WiFi.SSID(indices[j])) && (cschn == WiFi.channel(indices[j]))) {
-              DEBUG_CORE_LOG(PSTR(D_LOG_WIFI D_DUPLICATE_ACCESSPOINT " %s"), WiFi.SSID(indices[j]).c_str());
-              indices[j] = -1;  // set dup aps to index -1
-            }
-          }
-        }
-
         //display networks in page
-        for (uint32_t i = 0; i < n; i++) {
-          if (-1 == indices[i]) { continue; }  // skip dups
+        uint32_t i = 0;
+        while (i < n) {
           int32_t rssi = WiFi.RSSI(indices[i]);
+          String ssid = WiFi.SSID(indices[i]);
           DEBUG_CORE_LOG(PSTR(D_LOG_WIFI D_SSID " %s, " D_BSSID " %s, " D_CHANNEL " %d, " D_RSSI " %d"),
-            WiFi.SSID(indices[i]).c_str(), WiFi.BSSIDstr(indices[i]).c_str(), WiFi.channel(indices[i]), rssi);
-          int quality = WifiGetRssiAsQuality(rssi);
-          int auth = WiFi.encryptionType(indices[i]);
-          char encryption[20];
-          WSContentSend_P(PSTR("<div><a href='#p' onclick='c(this)'>%s</a>&nbsp;(%d)&nbsp<span class='q'>%s %d%% (%d dBm)</span></div>"),
-            HtmlEscape(WiFi.SSID(indices[i])).c_str(),
-            WiFi.channel(indices[i]),
-            GetTextIndexed(encryption, sizeof(encryption), auth +1, kEncryptionType),
-            quality, rssi
+            ssid.c_str(), WiFi.BSSIDstr(indices[i]).c_str(), WiFi.channel(indices[i]), rssi);
+
+          // Print SSID
+          WSContentSend_P(PSTR("<div><a href='#p' onclick='c(this)'>%s</a><br><ul>"),
+            HtmlEscape(ssid).c_str()
           );
-          delay(0);
+
+          int j = 0;
+          String nextSSID = "";
+          // Handle all APs with the same SSID
+          while((i+j) < n && (nextSSID = WiFi.SSID(indices[i+j])) == ssid) {
+            // Update RSSI / quality
+            rssi = WiFi.RSSI(indices[i+j]);
+            int quality = WifiGetRssiAsQuality(rssi);
+            // Color-code quality
+            uint8_t colors[2] = { 0xFF, 0xFF };
+            if(quality > 50) {
+              // Scale red component to go from yellow to green (full green)
+              colors[0] = (0xFF * (100-quality))/50;
+            } else {
+              // Scale green component to go from red to yellow (full red)
+              colors[1] = (0xFF * quality)/50;
+            }
```

Comment:
You can save even more here:

```
            if(quality > 50) {
              // Scale red component to go from yellow to green (full green)
              colors[0] = changeUIntScale(quality, 50, 100, 0xFF, 0);
            } else {
              // Scale green component to go from red to yellow (full red)
              colors[0] = changeUIntScale(quality, 0, 50, 0, 0xFF);
            }
```

I would also suggest to avoid pure Yellow #FFFF00 which is not readable with light background. Maybe going for a darker version would still do it:

```
            uint8_t colors[2] = { 0xAA, 0xAA };
            if(quality > 50) {
              // Scale red component to go from yellow to green (full green)
              colors[0] = changeUIntScale(quality, 50, 100, 0xAA, 0);
            } else {
              // Scale green component to go from red to yellow (full red)
              colors[0] = changeUIntScale(quality, 0, 50, 0, 0xAA);
            }
```

Reply:
I figured that most (all?) will run Tasmota in the standard configuration which seems to be a black background. This seems to be a wrong assumption and getting colors what are visible for all (e.g. green or orange background) is hard if not impossible. 

Removing the quality colors isn't much of a loss as the results are anyway sorted by RSSI (= quality) and I still think my improvement brings advantage to the display. Removing the color code, we're down to 597492 bytes so even ***less*** (8 bytes) than current `development`.

Reply:
Alternative - and color-independent - representation would be using HTML5 progress-bar
![Screenshot from 2020-12-26 09-14-21](https://user-images.githubusercontent.com/16748619/103148091-5dba8400-475c-11eb-9601-e1c7ee2d80b1.png)

I haven't pushed this so far, but it'd come at `597536` bytes (`+36` compared to `development`). As said before, my change without color or progress-bar comes at `597542` bytes (`-8` compared to `development`).